caddyconfig/caddyfile/lexer.go

```diff
@@ -168,3 +169,23 @@ func (l *lexer) next() bool {
 		val = append(val, ch)
 	}
 }
+
+// Tokenize takes bytes as input and lexes it into
+// a list of tokens that can be parsed as a Caddyfile.
+// Also takes a filename to fill the token's File as
+// the source of the tokens, which is important to
+// determine relative paths for `import` directives.
+func Tokenize(input []byte, filename string) ([]Token, error) {
+	l := lexer{}
+	if err := l.load(bytes.NewReader(input)); err != nil {
+		return nil, err
+	}
+	var tokens []Token
+	for l.next() {
+		if filename != "" {
+			l.token.File = filename
+		}
```

Comment:
Doesn't this just happen anyway? (regardless of `if`)

Reply:
Then they would be empty string instead of nil which seems like a relevant difference, no?

If not then sure I can remove the condition (or feel free to fix the PR yourself since I'm away for a bit)

Reply:
`filename` is a string, it can't be nil.